pyfr/plugins/turbulence.py

```diff
@@ -0,0 +1,272 @@
+from collections import defaultdict
+import math
+
+from pyfr.regions import BoxRegion
+import numpy as np
+
+from pyfr.plugins.base import BaseSolverPlugin
+from rtree.index import Index, Property
+
+
+def pcg(seed):
+    # pcg32rxs_m_xs
+    state = np.uint32(seed)
+    multiplier = np.uint32(747796405)
+    increment = np.uint32(2891336453)
+    mcgmultiplier = np.uint32(277803737)
+    n4, n22, n28 = np.uint32(4), np.uint32(22), np.uint32(28)
+    with np.errstate(over='ignore'):
+        while True:
+            ostate = ostatet = state
+            state = ostatet*multiplier + increment
+            rshift = np.uint32(ostatet >> n28)
+            ostatet ^= ostatet >> (n4 + rshift)
+            ostatet *= mcgmultiplier
+            ostatet ^= ostatet >> n22
+            random = (ostatet >> 8) * pow(2, -24)
+            yield (ostate, random)
+
+
+class TurbulencePlugin(BaseSolverPlugin):
+    name = 'turbulence'
+    systems = ['ac-navier-stokes', 'navier-stokes']
+    formulations = ['dual', 'std']
+    dimensions = [3]
+
+    def __init__(self, intg, cfgsect):
+        super().__init__(intg, cfgsect)
+
+        self.fdptype = fdptype = intg.backend.fpdtype
+
+        ac = intg.system.name.startswith('ac')
+
+        self.nmvxinit = 6
+        self.nvmxtol = 0.1
+
+        self.tstart = intg.tstart
+        self.tbegin = intg.tcurr
+        self.tnext = intg.tcurr
+        self.tend = intg.tend
+
+        self.seed = self.cfg.getint(cfgsect, 'seed')
```

Comment:
If restarting, what happens if the seed changes? Are there some restart checks that we might want to perform?

Reply:
Thats a good point. Let me think ...

Reply:
One option could be to make seed have a default, and so in most cases its not set by user (and if they do set it then  its a 'user beware' scenario)?

Reply:
After speaking to Freddie we have decided to not expose the seed.